### PR TITLE
Added comments for build and debug in Eclipse Neon.3

### PR DESCRIPTION
**Pull request by chinhuang007**
Signed-off-by: Chin Huang <chhuang@us.ibm.com>

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -48,3 +48,16 @@ To build without executing tests:
...
+
+The steps above still work to build without executing tests.
+
+There will be a few build errors in Eclipse Package Explorer. To fix the errors and debug JanusGraph code in Eclipse (breakpoints, step into/over, etc):
+
+1. Add “<your_janusgraph_home>/janusgraph-test/target/generated-sources/groovy-stubs/main” to the java build path for janusgraph-test project
+2. Change maven’s errors/warnings “Plugin execution not covered by lifecycle configuration” setting to “Ignore”
+3. Comment out the dependency for janusgraph-hadoop in the pom.xml for janusgraph-all project
+4. Update the java build path for your client java project, adding all JanusGraph java projects as required and removing JanusGraph jars from libraries
+
```
Comment (mbrukman) [inline BUILDING.md]: Drop blank line.

File: BUILDING.md [hunk 2]
```diff
@@ -48,3 +48,16 @@ To build without executing tests:
...
 * Windows: `for %i in (java.exe) do @echo. %~$PATH:i`
+
+--------------------------
+A new test is done on Eclipse Neon.3 Release (4.6.3) which already includes the m2e plugins.
+
+The steps above still work to build without executing tests.
+
+There will be a few build errors in Eclipse Package Explorer. To fix the errors and debug JanusGraph code in Eclipse (breakpoints, step into/over, etc):
+
+1. Add “<your_janusgraph_home>/janusgraph-test/target/generated-sources/groovy-stubs/main” to the java build path for janusgraph-test project
```
Comment (mbrukman) [inline BUILDING.md]: Use code formatting with backticks: `....` rather than double-quotes: "..."

Comment (mbrukman) [inline BUILDING.md]: Capitalize "Java" everywhere.

File: BUILDING.md [hunk 3]
```diff
@@ -48,3 +48,16 @@ To build without executing tests:
...
+2. Change maven’s errors/warnings “Plugin execution not covered by lifecycle configuration” setting to “Ignore”
```
Comment (mbrukman) [inline BUILDING.md]: Capitalize "Maven"

Comment (mbrukman) [inline BUILDING.md]: Use simple quotes rather than "smart quotes"

File: BUILDING.md [hunk 4]
```diff
@@ -48,3 +48,16 @@ To build without executing tests:
...
+3. Comment out the dependency for janusgraph-hadoop in the pom.xml for janusgraph-all project
```
Comment (mbrukman) [inline BUILDING.md]: Surround filenames and project names in backticks so they get code-formatted.

File: BUILDING.md [hunk 5]
```diff
@@ -48,3 +48,16 @@ To build without executing tests:
 To find the Java binary in your environment, run the appropriate command for your operating system:
 * Linux/macOS: `which java`
 * Windows: `for %i in (java.exe) do @echo. %~$PATH:i`
+
+--------------------------
+A new test is done on Eclipse Neon.3 Release (4.6.3) which already includes the m2e plugins.
```
Comment (mbrukman) [inline BUILDING.md]: This is not quite grammatically correct: "A new test is done on ..." — can you please clarify what you're trying to say?

File: BUILDING.md [hunk 6]
```diff
@@ -48,3 +48,16 @@ To build without executing tests:
 To find the Java binary in your environment, run the appropriate command for your operating system:
 * Linux/macOS: `which java`
 * Windows: `for %i in (java.exe) do @echo. %~$PATH:i`
+
+--------------------------
```
Comment (mbrukman) [inline BUILDING.md]: Drop this line; maybe add a new heading here instead?

```
### Neon.3 Release
```

File: BUILDING.md (hunk 1 above)
Comment (chinhuang007) [inline BUILDING.md]: Will make all the recommended changes.

**Comment by chinhuang007** [comment]
Sorry, I accidentally created multiple commits, intended to fix the original commit. Please let me know if I should abort this pull request and create a new one.

**Comment by pluradj** [comment]
@chinhuang007 You can squash all of your commits into a single commit, then force the push on your branch.

File: BUILDING.md [hunk 7]
```diff
@@ -48,3 +48,15 @@ To build without executing tests:
...
+## Neon.3 Release
+Note that the following has been tested on Eclipse Neon.3 Release (4.6.3). Neon.3 already includes the required m2e and m2e-wtp plugins.
+
+The steps above still work to build without executing tests.
+
+There will be a few build errors in Eclipse Package Explorer. To fix the errors and debug JanusGraph code in Eclipse (breakpoints, step into/over, etc):
+
+1. Add `<your_janusgraph_home>/janusgraph-test/target/generated-sources/groovy-stubs/main` to the Java build path for `janusgraph-test` project
+2. Change Maven’s errors/warnings "Plugin execution not covered by lifecycle configuration" setting to "Ignore"
+3. Comment out the dependency for `janusgraph-hadoop` in the `pom.xml` for `janusgraph-all` project
```
Comment (pluradj) [inline BUILDING.md]: I haven't tried it out for myself yet with Eclipse, but commenting out a dependency seems like a hack -- a hack that could likely get checked in by mistake. Any insight on why that particular dependency is messing up Eclipse?

**Pull request closed without merging**